### PR TITLE
Fix computation of when learning should be retried with lower learning rate.

### PR DESCRIPTION
**Pull request by alecw**
Hi @sjfleming ,

The code I added to implement --final-elbo-fail-fraction and --epoch-elbo-fail-fraction didn't really make sense before.  Now the percentage change is computed by comparing to (best_test_ELBO - first_train_ELBO) and (previous_test_ELBO - first_train_ELBO), respectively.  Basically, is there a big percentage change relative to the range of the ELBO plot.

In addition, I believe because miniconda is now python 3.8, I needed to add a version requirement for pytorch in order to not get an error.

Regards, Alec

### CODE REVIEW

**Comment by alecw** [comment]
Hi @sjfleming 

Please don't merge this yet.  I think new version of miniconda is causing packager version conflicts.  I'm investigating.

Regards, Alec

**Comment by alecw** [comment]
Hi @sjfleming ,

This should be good to go now.  I got a little unlucky because there was a problem with the 'packaging' module in which it caused a version requirement conflict, but they released a new version yesterday that fixed the problem.

-Alec

**Comment by alecw** [comment]
Hi @sjfleming , 

Will this PR ever be merged, or do we need to abandon this feature?

Regards, Alec

**Comment by sjfleming** [comment]
@alecw , sorry!  I had totally let this slip from my radar.  Yes I like this feature, and I think this is fine to merge, so I will do so now!

**Comment by sjfleming** [comment]
@alecw Do you want me to release a new version that includes this fix?

(We are hoping to make a bunch of pretty big changes and release them as v0.3.0 by early next year, so I could release this now as v0.2.2 if you wanted to use it with a version number.)

**Comment by alecw** [comment]
Thanks!  Does a new release include a new Docker image?  If so, a release would be great, because otherwise we have to deploy a private image.  Not a huge deal, but nice not to have to do.

Regards, Alec

**Comment by sjfleming** [comment]
Yeah I will make a docker image to go with it.  I will give that a shot later on today!

**Comment by sjfleming** [comment]
Alright, the docker image for the `v0.2.2` release is now ready and hosted at
```
us.gcr.io/broad-dsde-methods/cellbender:0.2.2
```
I gave it a test run and it seems to be in working order.
image
